[F]  Editors can manage project events in the backend

SMaxOwok opened the pull request:
[Fixes #285]

zdavis commented:
There are a number of UX issues that need to be addressed before this is ready for a code review.

1. Entering anything into the search field breaks the component entirely (see below). I don't think we want to index these events in ElasticSearch — simple text matching in PG should be sufficient for searching here.

<img width="857" alt="screen shot 2017-04-24 at 6 06 26 am" src="https://cloud.githubusercontent.com/assets/255909/25338477/3a7731d4-28b4-11e7-9cbe-231e7e95f896.png">

2. Vertical alignment in the list is weird. In fact, I don't see why we should use the same project list style here. I think it makes more sense in this case to list these out as event cards, just like how they're listed on the project page. We'll need to change the colors a bit to fit into the backend dark look, but let's use the same styles here.

<img width="518" alt="screen shot 2017-04-24 at 6 06 12 am" src="https://cloud.githubusercontent.com/assets/255909/25338528/63a21010-28b4-11e7-834c-8dffe1aa265c.png">

3. This won't matter if you use the cards, but with the existing layout, the relative sizing of the icons is inconsistent. The box looks much smaller than the egg.

<img width="978" alt="screen shot 2017-04-24 at 6 05 50 am" src="https://cloud.githubusercontent.com/assets/255909/25338692/f5d84a62-28b4-11e7-8153-1d21468da01d.png">

4. Tags are showing up in the list. Again, this should be addressed if we switch to the existing card layout.

<img width="71" alt="screen shot 2017-04-24 at 6 06 01 am" src="https://cloud.githubusercontent.com/assets/255909/25338670/e164072e-28b4-11e7-820b-acdd04189808.png">

SMaxOwok commented:
@zdavis this is using the existing event tiles in the list now.

zdavis commented:
Max,

This looks great and is getting really close. We need @naomiyaki to do a quick review on the CSS, and I noticed a few things which should be fixed:

1. We should have human readable strings in the filters dropdown rather than constants. You'll need to store the mapping between the constant and the string somewhere, and a component is likely not the best spot for it. For now it could be stored in src/config.js, which you can import into a component. We could add a new `locale` hash under `app`, and then include the mapping there.

![screen shot 2017-04-25 at 7 57 14 am](https://cloud.githubusercontent.com/assets/255909/25392701/378cdc46-298e-11e7-869e-248d24b46ef6.png)

2. The icon alignment is off in the screenshot below. The icon should be vertically centered with the text.

![screen shot 2017-04-25 at 7 57 01 am](https://cloud.githubusercontent.com/assets/255909/25392722/46264152-298e-11e7-8d3c-20e868cdb9bb.png)

3. I noticed what might be a regression in the paginated events list on the frontend (http://manifold.dev/browse/project/:id/events). In my environment, when I got to the all events view for metagaming and try to use the "next" button in pagination, it always takes me back to the 2nd page for some reason. See if you can reproduce it.

naomiyaki commented:
I love how this looks! Aside from Zach's comments above, I just have two small points of adjustment. 

- I really like the way the trashcan looks/is positioned, because it's absolutely positioned, there's a very very small possibility that the event text can run into it. This can be avoided by putting a bit more right padding on the `event-data` element in these lists only. 
![screen shot 2017-04-25 at 9 38 04 am](https://cloud.githubusercontent.com/assets/1628239/25397189/3d253032-299c-11e7-80f3-cb3d7cc40dc7.png)


- The `li` tags have a min height of 170, which makes sense on the front end where the tiles have tall buttons on the bottom, but seems a bit too tall here, maybe 140-150? @zdavis ? 
![manifold scholarship 2017-04-25 09-45-46](https://cloud.githubusercontent.com/assets/1628239/25397280/9f176b98-299c-11e7-9c19-1e1ab59a307a.png)

Thanks again!

zdavis commented:
Thanks, Naomi. Yeah, I think 140-150 could work for those.

SMaxOwok commented:
@zdavis, all the comments have been addressed.

- When the Searchable list tries to map the filter options, it will match to a set of labels, if provided.  I added the event types to the config file under "locale".
- Adjusted the vertical alignment of the bugle.
- Event page navigation was broken because I was missing a filter object in the request, so the page params were being passed as filters instead.
- Added a little more padding to the `event-data` in the tiles.
- Adjusted the size of the event tiles.

zdavis commented:
Hey Max — piece of advice. Anytime you're working with paginated result sets where it's possible to change the set (eg by deleting an event), see what happens when you remove all the results on a given page. In this case, the pagination goes all out of whack. This isn't a bug that's introduced by your code, and I'm fixing it, but this is a common sort of error, and worth keeping in your testing toolbelt.

zdavis commented on client/src/components/backend/Event/ListItem.js:
```diff
@@ -0,0 +1,42 @@
+    const attr = event.attributes;
+    const content = attr.excerpt ? attr.excerpt : attr.subjectTitle;
+    return (
+      <li className="list-item">
```
You need a unique key on this <li> (which I've added). Remember, any time you're rendering a collection like this in React, the wrapper element on each item needs a key.
